lib/fs_utils/pipeline.js

```diff
@@ -18,37 +18,25 @@ const prepareError = (type, stringOrError) => {
     stringOrError;
   const error = new Error(string);
   error.code = type;
-  return Promise.reject(error);
+  throw error;
 };
 
-
 // Run all linters.
 const lint = (data, path, linters) => {
-  const reduceLinter = (previousPromise, linter) => {
+  return linters.reduce((previousPromise, linter) => {
     debug(`Linting '${path}' with '${linter.constructor.name}'`);
     return previousPromise.then(() => {
       return callPlugin(linter, linter.lint, 2, [data, path]);
-    }, err => {
-      return Promise.reject(err);
-    });
-  };
-
-  return Promise.resolve(linters)
-    .then(linters => {
-      return linters.reduce(reduceLinter, Promise.resolve());
-    }).then(() => {
-      return Promise.resolve(data);
-    }, error => {
-      if (error.toString().match(warningRe)) {
-        logger.warn(`Linting of ${path}: ${error}`);
-        return Promise.resolve(data);
-      } else {
-        return prepareError('Linting', error);
-      }
     });
+  }, Promise.resolve()).then(() => data, error => {
```

Comment:
Yes, short arrow functions must be wrapped in parens only if they contain `{` or `,`